### PR TITLE
Add title replacement capability

### PR DESCRIPTION
**Pull request by rocketraman**
Fixes issue #4 . It isn't one or the other anymore -- the title can both be set and prefixed, or only one or the other.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -36,17 +38,30 @@ async function run() {
...
 
-    const titlePrefix = inputs.titleTemplate.replace(tokenRegex, match(inputs.uppercaseTitle));
+    const title = github.context.payload.pull_request.title;
+
+    const setTitle = inputs.titleReplaceTemplate ? inputs.titleReplaceTemplate.replace(tokenRegex, match(inputs.uppercaseTitle)) : null;
+    core.debug(`setTitle: ${setTitle}`);
+
+    const setTitle = setTitle && title !== setTitle
+    if (setTitle) {
+      request.title = setTitle;
```
Comment (tzkhan) [inline index.js:48]: I'm a bit confused here. I think this `setTitle` would be a `bool` here?

File: index.js [hunk 2]
```diff
@@ -36,17 +38,30 @@ async function run() {
...
+      core.warning('PR title set not requested or set already - no updates made');
+    }
+
+    const configuredTitleTemplate = inputs.titlePrefixTemplate || inputs.titleTemplate;
+    const titlePrefix = configuredTitleTemplate ? configuredTitleTemplate.replace(tokenRegex, match(inputs.uppercaseTitle)) : null;
     core.debug(`titlePrefix: ${titlePrefix}`);
 
-    const title = github.context.payload.pull_request.title;
-    const updateTitle = !title.toLowerCase().startsWith(titlePrefix.toLowerCase());
+    const prefixTitle = titlePrefix && !(request.title || title).toLowerCase().startsWith(titlePrefix.toLowerCase());
```
Comment (tzkhan) [inline index.js:58]: Actually, thinking about this, enabling *both* prefixing *and/or* replacing the title would not only complicate reasoning about the code here but also testing it. I can't think of a scenario where you'd need to specify templates for doing both. Wouldn't replacing the title make prefixing redundant?

**Comment by rocketraman** [comment]
All good feedback. I'll you're the pr and simplify it.

**Comment by tzkhan** [comment]
I've added a new release v1.1 that includes two new input properties - `replace-title` and `replace-body` that should help out with your requirement. Simply point the action reference to `uses: tzkhan/pr-update-action@v1.1` (you can even drop the version number) and you should be good to go.

**Comment by rocketraman** [comment]
Thanks @tzkhan .

**Pull request closed without merging**